File: web/models/mail_thread.ex
defmodule Firebrick.MailThread do
  use Ecto.Model

  schema "mail_threads" do
    field :subject, :string
    field :mail_label_ids, {:array, :integer}
    field :participant_ids, {:array, :integer}

    belongs_to :user, Firebrick.User
    has_many :mails, Firebrick.Mail
  end
end

File: frontend/app/models/mail-thread.js
import DS from 'ember-data';

export default DS.Model.extend({
  subject: DS.attr('string'),

  user: DS.belongsTo('user'),
  participants: DS.hasMany('contact'),
  mails: DS.hasMany('mail'),
  labels: DS.hasMany('mail-labels'),

  createdAt: DS.attr('date'),
  updatedAt: DS.attr('date')
});
